packages/contract/src/pay/DaimoPayRelayer.sol

```diff
@@ -91,30 +130,34 @@ contract DaimoPayRelayer is Ownable2Step {
             address(this)
         ) - amountPreSwap;
 
-        // Check the amount output from the swap. The contract owner tips the
-        // difference if needed. If there are excess tokens, transfer them to
-        // the owner.
+        //////////////////////////////////////////////////////////////
+        // POST-SWAP
+        //////////////////////////////////////////////////////////////
+
+        // If we received less than required, check that the amount we need to
+        // tip is within maxPostTip.
         if (swapAmountOut < requiredTokenOut.amount) {
```

Comment:
so if i understand correctly:

- during `start`, we have some amount of (random token) and need to produce an exact amount of the bridge token. maxPreTip is 0, maxPostTip is the max amount of the bridge token (eg USDC) we're willing to tip.

- during `fastfinish`/`claim` (whichever is actually running the intent) we have some amount of bridge token and we need an exact amount of destination token. maxPostTip is now 0, and maxPreTip is whatever amount of the bridge token we're willing to tip.

Reply:
correct